pkg/chains/objects/objects.go

```diff
@@ -58,6 +64,7 @@ type TektonObject interface {
 	GetLatestAnnotations(ctx context.Context, clientSet versioned.Interface) (map[string]string, error)
 	Patch(ctx context.Context, clientSet versioned.Interface, patchBytes []byte) error
 	GetResults() []Result
+	GetRefSource() *RefSource
```

Comment:
Prefer limiting things in this interface to the common denominator - in this case, the provenance field would be more fitting.

Both TaskRuns/PipelineRuns use the same type (and the same underlying RefSource type) so I don't think we'd need to define our own fork type here.

Reply:
fixed!

Reply:
I think we can go further - provenance feels like a common enough thing that we'll use for both pipelineruns and taskruns that we should return the type itself rather than make individual methods for its sub-fields.

Added suggestions!